### PR TITLE
chore: enable eslint react-hook/recommended rule set

### PR DESCRIPTION
**Pull request by guiccbr**
## Please verify the following:

- [x] `yarn test` **jest** tests pass with new tests, if relevant

## Describe your PR

Enable eslint react-hook/recommended rule set, as recommended in the react dev website: https://react.dev/learn/editor-setup#linting

Related discussion: https://github.com/infinitered/ignite/discussions/2453

### CODE REVIEW

File: src/commands/new.ts [hunk 1]
```diff
@@ -1,3 +1,4 @@
+/* eslint-disable react-hooks/rules-of-hooks */
```
Comment (guiccbr) [inline src/commands/new.ts:1]: This file has a method that starts with `use*` and react-hooks/rules-of-hooks is complaining about that.

I'm not sure if this is the best approach here.

File: package.json [hunk 2]
```diff
@@ -68,6 +68,7 @@
     "eslint-plugin-import": "^2.22.1",
     "eslint-plugin-node": "^11.1.0",
     "eslint-plugin-promise": "^5.1.0",
+    "eslint-plugin-react-hooks": "^4.6.0",
```
Comment (joshuayoes) [inline package.json:71]: Can you move this package to [boilerplate/package.json](https://github.com/infinitered/ignite/blob/master/boilerplate/package.json) instead? 

This top level package.json is for the cli, where as the `boilerplate` folder is for the app content.

File: package.json [hunk 3]
```diff
@@ -82,7 +83,8 @@
     "extends": [
       "prettier",
       "plugin:@typescript-eslint/recommended",
-      "standard"
+      "standard",
+      "plugin:react-hooks/recommended"
```
Comment (joshuayoes) [inline package.json:87]: Can you move this package to [boilerplate/package.json](https://github.com/infinitered/ignite/blob/master/boilerplate/package.json) instead?

This top level package.json is for the cli, where as the boilerplate folder is for the app content.

**Review by joshuayoes — CHANGES_REQUESTED** [review]
Made some comments to move some of the linting settings, but otherwise looks good so far

**Comment by guiccbr** [comment]
Thanks. I'll close and reopen because after enabling the the linter failed for the generated app. I'll need some more time to fix that.

**Pull request closed without merging**